packages/material-ui-codemod/src/v4.0.0/optimal-imports.js

```diff
@@ -0,0 +1,92 @@
+import { dirname } from 'path';
+import getJSExports from '../util/getJSExports';
+import addImports from 'jscodeshift-add-imports';
+
+// istanbul ignore next
+if (process.env.NODE_ENV === 'test') {
```

Comment:
We shouldn't publish env branches unless they compare with production.

You should be able to get the esModules index barrel by reading `@material-ui/core/package.json` and using the `module` entry. Then you can use `main` as a fallback to account for test runs in the monorepo.

Reply:
I don't understand about branches,  i'm checking if it's test mode, not if it's a branch

Reply:
Oh you mean a branch statement, not a code branch.

How do you want me to get the import path in testing?  `babel-plugin-module-resolver` doesn't work for for dynamic require.resolve statements.  Mocking require.resolve outside of this file won't work AFAIK.  I just don't see any other way

Reply:
Btw reading the package json and using it in require.resolve on subpaths still won't work in test mode.  Unlike the v3 codemod, this one parses the 2nd-level files to build whitelists, rather than the top-level file